db/init_db.js

```diff
@@ -1,31 +1,193 @@
 // code to build and initialize DB goes here
 const {
-  client
+  client,
   // other db methods 
 } = require('./index');
 
-async function buildTables() {
+const {createUser} = require('./db')
+const {createProduct} = require('./products')
+
+async function dropTables() {
+  client.connect();
+  console.log("Dropping Tables ...")
   try {
-    client.connect();
+    await client.query(`
+      DROP TABLE IF EXISTS order_products;
+      DROP TABLE IF EXISTS orders;
+      DROP TABLE IF EXISTS products;
+      DROP TABLE IF EXISTS users;
+    `);
+  } catch (error) {
+    throw error
+  }
+}
 
-    // drop tables in correct order
+async function buildTables() {
+  console.log("Building Tables ...")
+  try {
+    await client.query(`
+      CREATE TABLE users(
+        id SERIAL PRIMARY KEY,
+        "firstName" VARCHAR(255) NOT NULL,
+        "lastName" VARCHAR(255) NOT NULL,
+        email VARCHAR(255) UNIQUE NOT NULL,
+        imageURL VARCHAR(255) DEFAULT 'imageUrl',
+        username VARCHAR(255) UNIQUE NOT NULL,
+        password VARCHAR(255) UNIQUE NOT NULL,
+        "isAdmin" BOOLEAN DEFAULT false NOT NULL
+      );
+      CREATE TABLE products(
+        id SERIAL PRIMARY KEY,
+        name VARCHAR(255) NOT NULL,
+        description VARCHAR(255) NOT NULL,
+        price VARCHAR(255) NOT NULL,
+        imageURL VARCHAR(255) DEFAULT 'imageUrl',
+        "inStock" BOOLEAN DEFAULT false NOT NULL
+        category name VARCHAR(255) NOT NULL,
+      );
+      CREATE TABLE orders(
+        id SERIAL PRIMARY KEY,
+        status VARCHAR(255) DEFAULT 'created',
+        "userId" INTEGER REFERENCES users(id),
+        "datePlaced" DATE NOT NULL DEFAULT CURRENT_DATE
+      );
+      CREATE TABLE order_products(
+        id SERIAL PRIMARY KEY,
+        "productId" INTEGER REFERENCES products(id),
+        "orderId" INTEGER REFERENCES orders(id),
+        price VARCHAR(255) NOT NULL,
+        quantity INTEGER NOT NULL DEFAULT 0
+      );
```

Comment:
Should we include a UNIQUE constraint for ("productId", "orderId") to make sure we only get one instance of each product in a given order (especially since we have a quantity column)?

Reply:
This one confuses me a bit. So adding a unique constraint to productId and orderId would prevent duplicates of productId's within an order? And then same for order id, it would prevent duplicates of orderIds if that were to ever happen somehow? 

I couldn't tell from reading up on unique if it affects quantity? Like if someone orders 2 of an item the order would technically have 2 of that productId (aka a necessary duplicate)? Or am I confusing the logic behind this?

Reply:
Adding UNIQUE ("productId", "orderId") to this table prevents any two rows from having the same product and order ID combination. So within order #5, for example, you can't have two product #74s, only one. If we have product #74 in our cart, we can just change its quantity to add more.

Reply:
Ok! Gotcha i wasn't sure if setting a unique restraint could indirectly affect the concept of quantity for a product. I will throw those on there now!